### PR TITLE
Add AddPathPlugin support. ISSUE-255

### LINKED ISSUE #255: Include support for AddPathPlugin with just some configuration
| Q            | A
| ------------ | ---
| Bug?         | no
| New Feature? | yes

#### Actual Behavior

I needed to add a base path for all the requests done with a specific client. Httplug has support for this feature with AddPathPlugin but the HttplugBundle does not include it AFAIK (as it does for AddHostPlugin). For now I've followed a similar solution as done in #110 (thanks @dbu BTW :D).

```
services:
    app.http.path_uri:
        class: Psr\Http\Message\UriInterface
        factory: ["@httplug.uri_factory", createUri]
        arguments: ["%acme_client.api_path%"]

    app.http.plugin.add_path:
        class: Http\Client\Common\Plugin\AddPathPlugin
        arguments: ["@app.http.path_uri"]
```

```
httplug:
    clients:
        acme:
            factory: 'httplug.factory.guzzle6'
            plugins:
                - 'app.http.plugin.add_path'
```

#### Expected Behavior

Be able to add a base path with just some configuration:

```
httplug:
    clients:
        acme:
            factory: 'httplug.factory.guzzle6'
            plugins:
                - add_path:
                    path: '%acme_client.api_path%'
```

#### Possible Solutions

Add AddPathPlugin to HttplugBundle similar to AddHostPlugin.

### PR DESCRIPTION
**Pull request by enekochan**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #255
| Documentation   | ToDo
| License         | MIT

#### What's in this PR?

Add AddPathPlugin support

#### Why?

#255

#### Example Usage

http://docs.php-http.org/en/latest/plugins/request-uri-manipulations.html

#### Checklist

- [x] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [x] Documentation pull request created https://github.com/php-http/documentation/pull/229

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 The change log describes what is "Added", "Removed", "Changed" or "Fixed" between each release.
 
+## 1.9.1 (unreleased) - 2018-03-09
```
Comment (dbu) [inline CHANGELOG.md]: new features go into new minor versions. this should be 1.10.0

**Comment by dbu** [comment]
please also start the documentation pull request, i think the configuration will look exactly as you propose here.

**Comment by enekochan** [comment]
Done the semver version fix and the PR in de docs :) Thanks to @llaakkkk for the job in the base_uri PR as this is HEAVILY based on it ;)

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -298,6 +298,22 @@ private function createClientPluginNode()
...
+                    ->canBeEnabled()
+                    ->addDefaultsIfNotSet()
+                    ->info('Add a base path to the request.')
+                    ->children()
+                        ->scalarNode('path')
+                            ->info('Path to be added, e.g. /api/v1')
+                            ->isRequired()
+                            ->cannotBeEmpty()
+                        ->end()
+                        ->scalarNode('replace')
```
Comment (dbu) [inline DependencyInjection/Configuration.php]: are you sure about this one? looking at https://github.com/php-http/client-common/blob/master/src/Plugin/AddPathPlugin.php i think there is only the uri argument.

a request always has a path, so i think for path this just always prepends a path when configured, and there is no `replace` argument.

File: DependencyInjection/HttplugExtension.php [hunk 3]
```diff
@@ -207,6 +207,15 @@ private function configurePluginByName($name, Definition $definition, array $con
                     'replace' => $config['replace'],
                 ]);
 
+                break;
+            case 'add_path':
+                $pathUriService = $serviceId.'.path_uri';
+                $this->createUri($container, $pathUriService, $config['path']);
+                $definition->replaceArgument(0, new Reference($pathUriService));
+                $definition->replaceArgument(1, [
+                    'replace' => $config['replace'],
```
Comment (dbu) [inline DependencyInjection/HttplugExtension.php]: there is no argument 1 to the AddPath plugin

File: Resources/config/plugins.xml [hunk 4]
```diff
@@ -34,6 +34,10 @@
             <argument/>
             <argument/>
         </service>
+        <service id="httplug.plugin.add_path" class="Http\Client\Common\Plugin\AddPathPlugin" public="false" abstract="true">
+            <argument/>
+            <argument/>
```
Comment (dbu) [inline Resources/config/plugins.xml]: also remove this please

File: Tests/Unit/DependencyInjection/ConfigurationTest.php [hunk 5]
```diff
@@ -134,6 +134,13 @@ public function testSupportsAllConfigFormats()
                                 'replace' => false,
                             ],
                         ],
+                        [
+                            'add_path' => [
+                                'enabled' => true,
+                                'path' => '/api/v1',
+                                'replace' => false,
```
Comment (dbu) [inline Tests/Unit/DependencyInjection/ConfigurationTest.php]: and remove this

**Pull request merged**